expression/expression.go

```diff
@@ -97,13 +97,24 @@ type ReverseExpr interface {
 	ReverseEval(sc *stmtctx.StatementContext, res types.Datum, rType types.RoundingType) (val types.Datum, err error)
 }
 
+// TraverseDown implementing a pre-order traverse for expression tree.
+type TraverseDown interface {
+	Traverse(TraverseAction) Expression
+}
+
+// TraverseAction define the interface for action when traversing down an expression.
+type TraverseAction interface {
+	Transform(Expression) Expression
+}
+
 // Expression represents all scalar expression in SQL.
 type Expression interface {
 	fmt.Stringer
 	goJSON.Marshaler
 	VecExpr
 	ReverseExpr
 	CollationInfo
+	TraverseDown
```

Comment:
Seems that only `Expression` implements the `TraverseDown`. So we only need to
```suggestion
	Traverse(TraverseAction) Expression
```
?

Reply:
make sense